src/Raven.Server/Web/System/DatabasesHandler.cs

```diff
@@ -83,6 +86,83 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
+                var json = await context.ReadForDiskAsync(RequestBodyStream(), "Database Topology");
+                var databaseTopology = JsonDeserializationCluster.DatabaseTopology(json);
+
+                if (LoggingSource.AuditLog.IsInfoEnabled)
+                {
+                    var clientCert = GetCurrentCertificate();
+
+                    var auditLog = LoggingSource.AuditLog.GetLogger("DbMgmt", "Audit");
+                    auditLog.Info($"Database \'{dbName}\' record has been modified by {clientCert?.Subject} ({clientCert?.Thumbprint})");
+                }
+
+                // Validate Database Name
+                var databaseRecord = ServerStore.Cluster.ReadDatabase(context, dbName, out var index);
+                if (databaseRecord == null)
+                {
+                    throw new DatabaseDoesNotExistException($"Database {dbName} Record not found when trying to modify database topology");
+                }
+
+                // Validate Topology
+                var clusterTopology = ServerStore.GetClusterTopology(context);
+                var databaseAllNodes = databaseTopology.AllNodes;
+                foreach (var node in databaseAllNodes)
+                {
+                    if (clusterTopology.Contains(node) == false)
+                        throw new ArgumentException($"Failed to modify database {dbName} topology, because we don't have node {node} (which is in the new topology) in the cluster.");
+
+                    if (databaseRecord.Topology.RelevantFor(node))
+                    {
+                        var databaseIsBeenDeleted = databaseRecord.DeletionInProgress != null &&
+                                                    databaseRecord.DeletionInProgress.TryGetValue(node, out var deletionInProgress) &&
+                                                    deletionInProgress != DeletionInProgressStatus.No;
+                        if (databaseIsBeenDeleted)
+                            throw new InvalidOperationException($"Can't modify database {dbName} topology, because the database {dbName} is currently being deleted from node {node} (which is in the new topology)");
+
+                        var url = clusterTopology.GetUrlFromTag(node);
+                        if (url == null)
+                            throw new InvalidOperationException($"Can't modify database {dbName} topology, because node {node} (which is in the new topology) is not part of the cluster");
+
+                        if (databaseRecord.Encrypted && url.StartsWith("https:", StringComparison.OrdinalIgnoreCase) == false)
+                            throw new InvalidOperationException($"Can't modify database {dbName} topology, because database {dbName} is encrypted but node {node} (which is in the new topology) doesn't have an SSL certificate.");
+                    }
+                }
+                databaseTopology.ReplicationFactor = Math.Min(databaseTopology.Count, clusterTopology.AllNodes.Count);
```

Comment:
This looks like a code duplication from `AddDatabaseNode`, is that possible to reuse?

Reply:
No, this has a different logic, only the validation checks are a little bit alike, but not the same.
You need to update here the entire topology not only add nodes.

Reply:
Done